x-pack/plugins/spaces/server/routes/api/external/share_to_space.ts

```diff
@@ -74,4 +75,68 @@ export function initShareToSpacesApi(deps: ExternalRouteDeps) {
       return response.noContent();
     })
   );
+
+  const spacesSchema = schema.arrayOf(
+    schema.string({
+      validate: (value) => {
+        if (value !== ALL_SPACES_ID && !SPACE_ID_REGEX.test(value)) {
+          return `lower case, a-z, 0-9, "_", and "-" are allowed, OR "*"`;
+        }
+      },
+    }),
+    {
+      validate: (spaceIds) => {
+        if (uniq(spaceIds).length !== spaceIds.length) {
+          return 'duplicate space ids are not allowed';
+        }
+      },
+    }
+  );
+
+  externalRouter.post(
+    {
+      path: '/api/spaces/_update_objects_spaces',
```

Comment:
We didn't have any API Docs for the `_share_saved_object_add` and `_share_saved_object_remove` routes. Since we are moving away from Asciidocs, I plan to only make API Docs for the new `_get_shareable_references` and `_update_objects_spaces` routes using the new docs system. I'll add that in a follow-on PR along with other sharing-saved-objects-related guidance :+1: